uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -162,31 +153,117 @@ define([
     }());
 
     /**
-     * Calculate change in balance, it updates local balance variables.
-     *
-     * @return {Promise} empty promise if all required params are not set
-     *   otherwise promise from server
+     * Adjusts dates in UI.
+     * "To" date must not be earlier than "From" date in case of a Single day request.
+     * "To" date must be after "From" date in case of a Multiple day request.
      */
-    function calculateBalanceChange () {
-      vm._setDateAndTypes();
-      vm._toggleBalance();
+    function adjustDatesInUI () {
+      var toDate = moment(vm.uiOptions.toDate);
 
-      if (!vm._canCalculateChange()) { return $q.resolve(); }
+      if (vm.uiOptions.toDate) {
+        if (!vm.uiOptions.multipleDays) {
```

Comment:
@AkA84 Sorry, I still do not agree. 
`adjustDatesInUI` - adjusts because there is some logic that should never happen, and it sometimes, in edge cases, happens, especially when you change the absence period.
`setDatesFromUI` - sets dates from UI to the program. I would use `onUIDatesChange` but we discussed and decided not to use such names.

Reply:
> adjusts because there is some logic that should never happen, and it sometimes, in edge cases, happens, 

Well then let's see why it happens in the first place, seems this function is a symptom of a bigger problem

Reply:
I don't think there is "problem" at all here, when absence period is changed you need to be sure that dates are inside that period range.

Reply:
`adjustDatesInUI` is gone.